pulsar-broker/src/main/java/org/apache/pulsar/broker/loadbalance/extensions/data/BrokerLoadData.java

```diff
@@ -187,4 +191,35 @@ public String toString(ServiceConfiguration conf) {
         );
     }
 
+    public List<Metrics> toMetrics(String advertisedBrokerAddress) {
+        var metrics = new ArrayList<Metrics>();
+        var dimensions = new HashMap<String, String>();
+        dimensions.put("metric", "loadBalancing");
+        dimensions.put("broker", advertisedBrokerAddress);
```

Comment:
Normally, every metric will have a label to indicate the broker IP, do we need to add the label `broker`?

Reply:
This is for the backward compatibility as the existing resource usage metrics have the `broker` dimension. Operators can find the broker-id from the pod name too. For now, I don't see a particular reason we have to emit brokerUrl for all metrics.

Reply:
Sorry, I mean when we use Prometheus to collect metrics we can add some extra labels for the metric, such as Kubernetes pod name, etc.

Reply:
Yes, I believe those extra k8s labels can be automatically added later when pushing to Prometheus.

Reply:
It seems that the pod name is similar to this broker label.

Reply:
I see some metrics do have the "broker" dim and some do not in Pulsar now.

I added "broker" label for all metrics here.

Reply:
Sorry, there seems to be a misunderstanding, I just want to know if we can use the extra labels (such as k8s pod name) instead of this newly added broker label.

Reply:
Like I said, Pulsar brokers do not emit pod specific metrics. Some users do run on bare metal instances. IMHO, It would be a bad design to access k8s env vars from brokers.

Reply:
The k8s pod name is just an example, I don't want to collect k8s metrics from the broker. I wanted to add some extra labels when Prometheus scrapes, just like the label `instance`, but it's not ok, because the time series data may come from a proxy service, actually most broker metrics all need the broker dimension.

<img width="912" alt="figure-1" src="https://user-images.githubusercontent.com/15029908/217755435-f1b0a6e1-847c-42dc-9a6e-abd05ad413df.png">

Reply:
Like I said, I think such extra labels(instance and job labels) are automatically injected at the Prometheus layer.
I don't see Pulsar broker code specifying these labels. Please correct me if I am wrong here.

Reply:
Yes, they are automatically injected at the Prometheus.

Reply:
> What I want to express is that the instance label is similar to the broker dim.

Do you mean these metrics should not contain the `broker` dimension because `broker` is redundant?

> This is for the backward compatibility as the existing resource usage metrics have the broker dimension.

As I said above, some existing pulsar metrics explicitly emit this `broker` dimension, and I kept them here for the backward compatibility. 

Since I already added the `broker` dim for all metrics here. At least we are consistent.

Reply:
@heesung-sn 

> However, as I said above, some existing pulsar metrics explicitly emit this broker label, and I kept them here for the backward compatibility.

Could you please give more context about this one? I haven't got the key point of what is the compatibility issue that we need to resolve. We don't have the `broker` label for all the metrics that exposed by broker. And it will be duplicated with `instance` label for most cases. So we need a strong reason to add the `broker` label to show it is necessary and  the `instance` label will not be a good substitute here.

Reply:
If there is a compatibility problem, I think it's ok.

Reply:
@codelipenghui 
I see the following metrics currently emit `broker` label.
```
curl http://localhost:8080/metrics/ |grep broker=
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 99144    0 99144    0     0  1739k      0 --:--:-- --:--:-- --:--:-- 2060k
pulsar_active_connections{cluster="standalone", broker="localhost", metric="broker_connection"} 2
pulsar_connection_closed_total_count{cluster="standalone", broker="localhost", metric="broker_connection"} 0
pulsar_connection_create_fail_count{cluster="standalone", broker="localhost", metric="broker_connection"} 0
pulsar_connection_create_success_count{cluster="standalone", broker="localhost", metric="broker_connection"} 2
pulsar_connection_created_total_count{cluster="standalone", broker="localhost", metric="broker_connection"} 2
pulsar_lb_bandwidth_in_usage{cluster="standalone", broker="localhost", metric="loadBalancing"} 0.0
pulsar_lb_bandwidth_out_usage{cluster="standalone", broker="localhost", metric="loadBalancing"} 0.0
pulsar_lb_cpu_usage{cluster="standalone", broker="localhost", metric="loadBalancing"} 0.0
pulsar_lb_directMemory_usage{cluster="standalone", broker="localhost", metric="loadBalancing"} 26.953125
pulsar_lb_memory_usage{cluster="standalone", broker="localhost", metric="loadBalancing"} 15.332031
```